patches/blockprotocol+0.0.7.patch

```diff
@@ -33,49 +33,160 @@ index 20a9bbd..a849fa3 100644
+ };
+@@ -304,14 +294,7 @@ export type BlockProtocolCreateLinksFunction = {
    (actions: BlockProtocolCreateLinksAction[]): Promise<BlockProtocolLink[]>;
  };
  
-+export type BlockProtocolUpdateLinkedAggregationActionFragment = {
-+  // temporary identifiers for LinkedAggregations - to be replaced with a single id
-+  sourceAccountId?: string | null;
-+  sourceEntityId: string;
-+  path: string;
-+}
-+
- export type BlockProtocolUpdateLinksAction =
-   | { linkId: string; data: Pick<BlockProtocolLink, "index"> }
+-export type BlockProtocolUpdateLinksAction =
+-  | { linkId: string; data: Pick<BlockProtocolLink, "index"> }
 -  | ({
 -      // temporary identifiers for LinkedAggregations - to be replaced with a single id
 -      sourceAccountId?: string | null;
 -      sourceEntityId: string;
 -      path: string;
 -    } & { data: BlockProtocolAggregateOperationInput });
-+  | (BlockProtocolUpdateLinkedAggregationActionFragment & { data: BlockProtocolAggregateOperationInput });
++export type BlockProtocolUpdateLinksAction = { linkId: string; data: Pick<BlockProtocolLink, "index"> }
  
  export type BlockProtocolUpdateLinksFunction = {
    (actions: BlockProtocolUpdateLinksAction[]): Promise<BlockProtocolLink[]>;
-@@ -452,4 +459,4 @@ export type BlockProtocolProps = {
-   linkedAggregations?: BlockProtocolLinkedAggregation[];
+@@ -327,6 +310,73 @@ export type BlockProtocolDeleteLinksFunction = {
+   (actions: BlockProtocolDeleteLinksAction[]): Promise<boolean[]>;
+ };
+ 
++// ---------------------- LINKED AGGREGATIONS ------------------------- //
++
++export type SingleTargetLinkFields = {
++  destinationEntityId: string;
++  destinationEntityTypeId?: string | null;
++  destinationEntityVersionId?: string | null;
++};
++
++export type BlockProtocolLinkedAggregationInput = {
++  aggregationId: string;
++  sourceAccountId?: string | null;
++  sourceEntityId: string;
++  sourceEntityTypeId?: string | null;
++  sourceEntityVersionId?: string | null;
++  path: string;
++  operation: BlockProtocolAggregateOperationInput;
++};
++
++export type BlockProtocolLinkedAggregation = 
++  Omit<BlockProtocolLinkedAggregation, "operation">
++  & BlockProtocolAggregateEntitiesResult<BlockProtocolEntity>;
++
++export type BlockProtocolGetLinkedAggregationAction = {
++  sourceAccountId: string;
++  aggregationId: string;
++};
++
++export type BlockProtocolGetLinkedAggregationsFunction = {
++  (actions: BlockProtocolGetLinkedAggregationAction[]): Promise<BlockProtocolLinkedAggregation[]>;
++};
++
++export type BlockProtocolCreateLinkedAggregationAction = DistributedOmit<
++  BlockProtocolLinkedAggregationInput,
++  "aggregationId"
++>;
++
++export type BlockProtocolCreateLinkedAggregationsFunction = {
++  (actions: BlockProtocolCreateLinkedAggregationAction[]): Promise<BlockProtocolLinkedAggregation[]>;
++};
++
++export type BlockProtocolUpdateLinkedAggregationActionFragment = {
++  sourceAccountId: string;
++  sourceEntityId: string;
++  aggregationId: string;
```

Comment:
```suggestion
+  sourceAccountId?: string | null;
+  sourceEntityId?: string | null;
+  aggregationId: string;
```

Reply:
You've explained your thinking on `sourceEntityId`, why should `sourceAccountId` be nullable? Is it just for the `blockprotocol` package BP method type declaration, whereas our implementation throws an error when `accountId` is passed (because we need it for sharding)?

Reply:
Yes - In the BP, the only identification field at the moment is entityId (and equivalents), because in many apps a single field may be sufficient to identify an entity. So we don't require any other id fields.

So it'll be optional in the BP types when we update them. In theory we could have it required in this patch and fix it later, but feels small enough to just do now

Reply:
Addressed in https://github.com/hashintel/hash/pull/497/commits/800b6af44e74458c53b78ff9a8321b97e53dbc46